.github/workflows/_reusable_helm.yml

```diff
@@ -0,0 +1,134 @@
+          FILENAME_FORMAT: "{kind}-{group}-{version}"
+        run: |
+          set -o pipefail
+          OS=$(uname)
+          # install kubeconform
+          curl -sSfL "${KUBECONFORM_BASE_URL}/${KUBECONFORM_VERSION}/kubeconform-${OS}-amd64.tar.gz" |
+          tar -xzf - kubeconform
+          # validate changed charts
+          for chart in ./helm/charts/*/; do
+            values=$(echo "${chart}" | sed "s/charts/values/g")
+            echo "Running kubeconform for folder: '${chart}'"
+            helm dep up "${chart}" &&
+            for value in $(find "${values}" -type f -name "*.yaml"); do
+              echo -e "▶ Validating ${chart} with values from ${value}"
+              helm template --kube-version "${KUBERNETES_VERSION#v}" -f "${value}" "${chart}" |
+              ./kubeconform -strict -ignore-missing-schemas -exit-on-error \
+                -schema-location default -schema-location 'helm/k8s-crds-schemas/{{ .ResourceKind }}{{ .KindSuffix }}.json' \
+                -kubernetes-version "${KUBERNETES_VERSION#v}" -summary -verbose -schema-location default
+            done
+          done
+
+  publish-chartmuseum-dev:
+    name: publish-chartmuseum-dev
+    environment: dev
+    runs-on: [self-hosted, shared]
+    needs: [lint-chart, lint-docs, kubeconform-chart]
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: Add chartmuseum-dev helm repo
+        run: |
+          if ! helm plugin list | grep -q push; then
+            helm plugin install https://github.com/chartmuseum/helm-push.git
+          fi
+          helm repo add chartmuseum-dev --username ${{ secrets.CHARTMUSEUM_DEV_USER }} --password ${{ secrets.CHARTMUSEUM_DEV_PASSWORD }} ${{ secrets.CHARTMUSEUM_DEV_URL }} &&
+          helm repo update
+      - name: Push chart to chartmuseum-dev repo
+        run: |
+          for chart in ./helm/charts/*/; do
+            helm cm-push -f $chart chartmuseum-dev --version ${{ inputs.version }}
+          done
+
+  publish-chartmuseum-prd:
+    name: publish-chartmuseum-prd
+    if: inputs.release != false
+    runs-on: [self-hosted, shared]
+    needs: [publish-chartmuseum-dev]
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: Add chartmuseum-prd helm repo
+        run: |
+          if ! helm plugin list | grep -q push; then
+            helm plugin install https://github.com/chartmuseum/helm-push.git
+          fi
+          helm repo add chartmuseum-prd --username ${{ secrets.CHARTMUSEUM_PRD_USER }} --password ${{ secrets.CHARTMUSEUM_PRD_PASSWORD }} ${{ secrets.CHARTMUSEUM_PRD_URL }} &&
+          helm repo update
+      - name: Push chart to chartmuseum-prd repo
+        run: |
+          for chart in ./helm/charts/*/; do
+            helm cm-push $chart chartmuseum-prd --version ${{ inputs.version }}
+          done
```

Comment:
This also means that prod credentials are injected in each and every jobs :/

Reply:
Yes, indeed, as DockerHub, Nexus and PYPI. 
We have a separated track with Sec team to secure our pipelines, we must address it.

Reply:
Ideally, I would prefer to work on "dev museum" for the integration, and copy on production only when we decided to "promote release". But the issue is that every environments including Minivault is using production...

Reply:
wdyt @cjudith-ledger ?

Reply:
As discussed and validated with @sergii-shkolin and @amanone,  

1. Continuing CI/CD migration where we push on dev & prd CM at release time and set the “secrets” at repo level (I will need admin permissions if it’s not the case). So, no need to set at the org. level.
2. Later, we will migrate all non prod envs to point on “non prod” CM
3. We migrate the “push” from the different repos to the “release” pipeline to promote charts from dev to prd.